CondTools/RunInfo/src/LHCInfoHelper.cc

```diff
@@ -0,0 +1,29 @@
+#include "CondTools/RunInfo/interface/LHCInfoHelper.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//*****************************************************************
+// Returns lumi-type IOV (packed using cond::time::lumiTime) from
+// last LS of last Run of the specified Fill
+//*****************************************************************
+cond::Time_t cond::lhcInfoHelper::getFillLastLumiIOV(const cond::OMSService& oms, unsigned short fillId) {
+  // Define query
+  auto query = oms.query("lumisections");
+  query->addOutputVars({"lumisection_number", "run_number"});
+  query->filterEQ("fill_number", fillId);
+  query->limit(cond::lhcInfoHelper::kLumisectionsQueryLimit);
+
+  // Execute query
+  if (!query->execute())
+    throw cms::Exception("OMSQueryFailure")
+        << "OMS query of fill " << fillId << " failed, status:" << query->status() << "\n";
+
+  // Get query result
+  auto queryResult = query->result();
+  if (queryResult.empty())
+    throw cms::Exception("OMSQueryFailure") << "OMS query of fill " << fillId << " returned empty result!\n";
+
+  // Return the final IOV
+  auto myLastRun = queryResult.back().get<int>("run_number");
+  auto myLastLumi = queryResult.back().get<unsigned short>("lumisection_number");
+  return cond::time::lumiTime(myLastRun, myLastLumi);
```

Comment:
also minor: I would change `myLastRun` and `myLastLumi` to  `lastRun` and `lastLumi`, the "my" is just unnecessary